core/slice.go

```diff
@@ -128,9 +131,9 @@ func (sl *Slice) Append(header *types.Header, domPendingHeader *types.Header, do
 	}
 
 	// Construct the block locally
-	block := sl.ConstructLocalBlock(header)
-	if block == nil {
-		return nil, ErrBodyNotFound
```

Comment:
we need this Body not found in the case of not synced. you cannot validate if there is no body and it will crash the node.

Reply:
d0aa3c4e please look at the change in this commit

Reply:
Good catch. Will fix :+1: